packages/@react-aria/grid/src/useGridCell.ts

```diff
@@ -205,18 +249,101 @@ export function useGridCell<T, C extends GridCollection<T>>(props: GridCellProps
         break;
       }
       case 'ArrowUp':
-      case 'ArrowDown':
+      case 'ArrowDown': {
+        if (keyboardNavigationBehavior === 'tab') {
+          break;
+        };
+
         // Prevent this event from reaching cell children, e.g. menu buttons. We want arrow keys to navigate
         // to the cell above/below instead. We need to re-dispatch the event from a higher parent so it still
         // bubbles and gets handled by useSelectableCollection.
-        if (!e.altKey && ref.current.contains(e.target as Element)) {
+        if (keyboardNavigationBehavior === 'arrow' && !e.altKey && ref.current.contains(e.target as Element)) {
           e.stopPropagation();
           e.preventDefault();
-          ref.current.parentElement.dispatchEvent(
-            new KeyboardEvent(e.nativeEvent.type, e.nativeEvent)
-          );
+
+          if (isEditing) {
+            ref.current.dispatchEvent(
+              new KeyboardEvent(e.nativeEvent.type, {...e.nativeEvent, key: e.key === 'ArrowUp' ? 'ArrowLeft' : 'ArrowRight'})
+            );
+          } else {
+            ref.current.parentElement.dispatchEvent(
+              new KeyboardEvent(e.nativeEvent.type, e.nativeEvent)
+            );
+          }
+        }
+        break;
+      }
+      case 'F2':
+      case 'Enter': {
+        if (isEditable) {
+          let focusable = walker.nextNode() as FocusableElement;
+
+          if (focusable && document.activeElement === ref.current) {
+            state.setEditingKey(node.key);
+            focusSafely(focusable);
+          }
+        }
+
+        if (isEditable && state.editingKey && (isSingleLineTextInput(document.activeElement) || e.key === 'F2')) {
+          state.setEditingKey(null);
+          focusSafely(ref.current);
+        }
+        break;
+      }
+      case 'Escape': {
+        if (state.editingKey || keyboardNavigationBehavior === 'tab') {
+          state.setEditingKey(null);
+          focusSafely(ref.current);
+        }
+        break;
+      }
+      case 'Tab': {
+        if (keyboardNavigationBehavior === 'tab') {
+          // If there is another focusable element within this item, stop propagation so the tab key
+          // is handled by the browser and not by useSelectableCollection (which would take us out of the list).
+          let walker = getFocusableTreeWalker(ref.current, {tabbable: true});
+          walker.currentNode = document.activeElement;
+          let next = e.shiftKey ? walker.previousNode() : walker.nextNode();
+
+          if (next || isEditing) {
+            e.stopPropagation();
+
+            if (isEditable && document.activeElement === ref.current) {
+              state.setEditingKey(node.key); // Do we want this?
```

Comment:
Not sure about this since it creates disparities between the two navigation methods.
I do find it nice to maintain the `Tab` to enter cell from grid list however.